src/basic/iovec-wrapper.h

```diff
@@ -22,16 +18,7 @@ void iovw_done_free(struct iovec_wrapper *iovw);
 void iovw_done(struct iovec_wrapper *iovw);
 
 int iovw_put(struct iovec_wrapper *iovw, void *data, size_t len);
-static inline int iovw_consume(struct iovec_wrapper *iovw, void *data, size_t len) {
-        /* Move data into iovw or free on error */
-        int r;
-
-        r = iovw_put(iovw, data, len);
-        if (r < 0)
-                free(data);
```

Comment:
free() is forward declared, so this seems unnecessary?

Reply:
Dropped